scintilla/src/Document.cxx

```diff
@@ -3265,11 +3265,14 @@ class UTF8Iterator {
 #if defined(SCI_OWNREGEX) && defined(BOOST_REGEX_STANDALONE)
 boost::regex_constants::match_flag_type MatchFlags(const Document* doc, Sci::Position startPos, Sci::Position endPos) noexcept {
 	boost::regex_constants::match_flag_type flagsMatch =
-		boost::regex_constants::match_default | boost::regex_constants::match_not_dot_newline;
+		boost::regex_constants::match_default
+		| boost::regex_constants::match_not_dot_newline
+		| boost::regex_constants::match_not_bol
+		| boost::regex_constants::match_not_eol;
```

Comment:
This make it worse, put caret before 1 on line start, it can't find 123:
```
123
```

Not sure whether worth it, `IsWordStartAt()` and `IsWordEndAt()` can be used to check word boundary.